src/lean/ffi/u128.rs

```diff
@@ -17,3 +17,25 @@ extern "C" fn rs_mul_u128_in_binary_field(a: &u128, b: &u128) -> *const u128 {
     let c = a * b;
     to_raw(c.val())
 }
+
+#[unsafe(no_mangle)]
+extern "C" fn rs_mul_u64_in_binary_field(a: u64, b: u64) -> u64 {
+    let a = B64::new(a);
+    let b = B64::new(b);
+    let c = a * b;
+    c.val()
+}
+
+#[unsafe(no_mangle)]
+extern "C" fn rs_pow_u64_in_binary_field(a: u64, b: u64) -> u64 {
+    let a = B64::new(a);
+    let c = a.pow([b]);
+    c.val()
+}
+
+#[unsafe(no_mangle)]
+extern "C" fn rs_inv_u64_in_binary_field(a: u64) -> u64 {
+    let a = B64::new(a);
+    let c = a.invert().unwrap_or(B64::new(0));
```

Comment:
This is often done by letting the underlying C/Rust code panic but you'd prevent `a` from being zero at the Lean level, by requiring a proof that it's not zero.